src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -1347,5 +1317,326 @@ private void VerifyNotepadMainWindowTitle(Process process, string filename)
+            }
+            catch (Win32Exception ex) when (ex.NativeErrorCode == ERROR_SHARING_VIOLATION)
+            {
+                throw new SkipTestException($"{process.StartInfo.FileName} has been locked by some other process");
+            }
+        }
+
+        private class TestProcessState : IDisposable
+        {
+            private readonly Process _process;
+
+            private readonly bool _hasStarted;
+
+            private readonly WindowsTestAccount _processAccount;
+
+            private readonly WindowsTestAccount _processImpersonationAccount;
+
+            private readonly string _workingDirectory;
+
+            private readonly Action<string, string> _additionalCleanup;
+
+            private bool _disposedValue;
+
+            public TestProcessState(
+                Process process,
+                bool hasStarted,
+                WindowsTestAccount processAccount,
+                WindowsTestAccount processImpersonationAccount,
+                string workingDirectory,
+                Action<string, string> additionalCleanup)
+            {
+                _process = process;
+                _hasStarted = hasStarted;
+                _processAccount = processAccount;
+                _processImpersonationAccount = processImpersonationAccount;
+                _workingDirectory = workingDirectory;
+                _additionalCleanup = additionalCleanup;
+            }
+
+            public string ProcessAccountName => _processAccount?.AccountName;
+
+            public string ImpersonationAccountName => _processImpersonationAccount?.AccountName;
+
+            protected virtual void Dispose(bool disposing)
+            {
+                if (!_disposedValue)
+                {
+                    if (disposing)
+                    {
+                        if (_hasStarted)
+                        {
+                            _process.Kill();
+
+                            Assert.True(_process.WaitForExit(WaitInMS));
+                        }
+
+                        if (PlatformDetection.IsNotWindowsServerCore)
+                        {
+                            // remove the access
+                            SetAccessControl((_processImpersonationAccount ?? _processAccount).AccountName, _process.StartInfo.FileName, _workingDirectory, add: false);
+                        }
+
+                        _additionalCleanup?.Invoke(_processAccount?.AccountName, _processImpersonationAccount?.AccountName);
+                        _processAccount?.Dispose();
+                        _processImpersonationAccount?.Dispose();
+                    }
+
+                    _disposedValue = true;
+                }
+            }
+
+            public void Dispose()
+            {
+                Dispose(disposing: true);
+                GC.SuppressFinalize(this);
+            }
```

Comment:
this type does not define a finalizer, so there is no need to call `GC.SuppressFinalize(this)`. We also don't expect anyone to derive from it, so there is no need for `virtual void Dispose(bool disposing)`

```suggestion
            public void Dispose()
            {
                if (_hasStarted)
                {
                    _process.Kill();

                    Assert.True(_process.WaitForExit(WaitInMS));
                }

                if (PlatformDetection.IsNotWindowsServerCore)
                {
                    // remove the access
                    SetAccessControl((_processImpersonationAccount ?? _processAccount).AccountName, _process.StartInfo.FileName, _workingDirectory, add: false);
                }

                _additionalCleanup?.Invoke(_processAccount?.AccountName, _processImpersonationAccount?.AccountName);
                _processAccount?.Dispose();
                _processImpersonationAccount?.Dispose();
            }
```

Reply:
This seems to be a somewhat subjective topic - some analyzers demand to always use the full disposable pattern with suppressfinalize, etc., while others accept the simplified form. Happy to use whatever you recommend for the project. 👍🏻